pkg/plugins/plugins_manager.go

```diff
@@ -0,0 +1,163 @@
+package plugins
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+// PluginGroup specifies the group name of related plugins
+// For example, "diff" is a group of plugins that know how to perform a diff.
+type PluginGroup string
+
+// PluginType specifies a type of plugin
+// For example, "delta" is a plugin.
+type PluginType string
+
+// PluginIdentity identifies the plugin using a Key and Secret, and supplies the Version it expects the plugin to be of.
+type PluginIdentity struct {
+	Version uint
+	Key     string
+	Secret  string
+}
+type PluginTypeConfigMap map[PluginType]plugin.ClientConfig
+
+// PluginGroupTypeMap example:
+/*
+{
+	"diff": {
+		"delta": plugin.ClientConfig{
+			HandshakeConfig: plugin.HandshakeConfig{
+				ProtocolVersion:  1,
+				MagicCookieKey:   "key",
+				MagicCookieValue: "value",
+			},
+			Cmd: exec.Command("path/to/executable/diff/delta"),
+			AllowedProtocols: []plugin.Protocol{
+				plugin.ProtocolGRPC,
+			},
+			Plugins: map[string]plugin.Plugin{
+				"deltaDiff": DeltaGRPCPlugin,
+			},
+		},
+		"iceberg: {
+			HandshakeConfig: plugin.HandshakeConfig{
+				ProtocolVersion:  1,
+				MagicCookieKey:   "key",
+				MagicCookieValue: "value",
+			},
+			Cmd: exec.Command("path/to/executable/diff/iceberg"),
+			AllowedProtocols: []plugin.Protocol{
+				plugin.ProtocolGRPC,
+			},
+			Plugins: map[string]plugin.Plugin{
+				"icebergDiff": IcebergGRPCPlugin
+			},
+		}
+	},
+	"merge": {
+		"delta": {
+			HandshakeConfig: plugin.HandshakeConfig{
+				ProtocolVersion:  1,
+				MagicCookieKey:   "key",
+				MagicCookieValue: "value",
+			},
+			Cmd: exec.Command("path/to/executable/merge/delta"),
+			AllowedProtocols: []plugin.Protocol{
+				plugin.ProtocolGRPC,
+			},
+			Plugins: map[string]plugin.Plugin{
+				"deltaMerge": DeltaGRPCPlugin,
+			},
+		},
+		"iceberg: {
+			HandshakeConfig: plugin.HandshakeConfig{
+				ProtocolVersion:  1,
+				MagicCookieKey:   "key",
+				MagicCookieValue: "value",
+			},
+			Cmd: exec.Command("path/to/executable/merge/iceberg"),
+			AllowedProtocols: []plugin.Protocol{
+				plugin.ProtocolGRPC,
+			},
+			Plugins: map[string]plugin.Plugin{
+				"icebergMerge": IcebergGRPCPlugin
+			},
+		}
+	},
+}
+*/
+type PluginGroupTypeMap map[PluginGroup]PluginTypeConfigMap
```

Comment:
Personal taste: don't really need the type definitions for the map types. Simple maps can work here, as these types are not exposed, if I understand correctly.